includes/admin/plugins.php

```diff
@@ -9,58 +9,53 @@
  * @since       1.8
  */
 
-// Exit if accessed directly
+// Exit if accessed directly.
 defined( 'ABSPATH' ) || exit;
 
 /**
  * Plugins row action links
  *
- * @author Michael Cannon <mc@aihr.us>
  * @since 1.8
- * @param array $links already defined action links
- * @param string $file plugin file path and name being processed
+ * @since 3.1 Targeted just our plugin.
+ *
+ * @param array $links Already defined action links.
+ *
  * @return array $links
  */
-function edd_plugin_action_links( $links = array(), $file = '' ) {
+function edd_plugin_action_links( $links = array() ) {
+	$edd_links = array();
+
+	$pass_manager = new EDD\Admin\Pass_Manager();
+	if ( ! $pass_manager->has_pass() ) {
 
-	// Only EDD plugin row
-	if ( EDD_PLUGIN_BASE === $file ) {
-		$settings_url = edd_get_admin_url( array(
-			'page' => 'edd-settings'
-		) );
+		$url = edd_link_helper(
+			'https://easydigitaldownloads.com/lite-upgrade/',
+			array(
+				'utm_medium'  => 'all-plugins',
+				'utm_content' => 'upgrade-to-pro',
+			)
+		);
 
-		$links['settings'] = '<a href="' . esc_url( $settings_url ) . '">' . esc_html__( 'Settings', 'easy-digital-downloads' ) . '</a>';
+		$edd_links['edd-pro-upgrade'] = sprintf( '<a href="%s" target="_blank">' . __( 'Upgrade to Pro', 'easy-digital-downloads' ) . '</a>', $url );
 	}
 
-	// Return array of links
-	return $links;
-}
-add_filter( 'plugin_action_links', 'edd_plugin_action_links', 10, 2 );
+	$settings_url = edd_get_admin_url(
+		array(
+			'page' => 'edd-settings',
+		)
+	);
+
+	$edd_links['settings'] = '<a href="' . esc_url( $settings_url ) . '">' . esc_html__( 'Settings', 'easy-digital-downloads' ) . '</a>';
 
+	// Return array of links.
+	return array_merge( $edd_links, $links );
+}
+add_filter( 'plugin_action_links_easy-digital-downloads/easy-digital-downloads.php', 'edd_plugin_action_links', 10, 2 );
 
 /**
- * Plugin row meta links
- *
- * @author Michael Cannon <mc@aihr.us>
- * @since 1.8
- * @param array $links already defined meta links
- * @param string $file plugin file path and name being processed
- * @return array $input
+ * Load any CSS we need for the plugins list table.
  */
-function edd_plugin_row_meta( $links = array(), $file = '' ) {
```

Comment:
Should this function definition be moved to the deprecated functions file?

Reply:
Fixed.